### ISSUE TITLE
Command failed (output redirection?)

### ISSUE THREAD

**Issue by krmaxwell** [nirvdrum/svn2git#132]
Got the following error:

```
$ svn2git http://bletchley.googlecode.com/svn/trunk/
command failed:
2>&1 git checkout -f master
```

**Comment by fenrhil** [comment]
With git 1.9.1 and svn2git 2.3.2, the bug mentioned in [post 3](https://github.com/nirvdrum/svn2git/issues/132#issuecomment-22940832) does not seem to be fixed. Maybe a regression?

I just ran `svn2git svn://svn.code.sf.net/p/jade-plugins/code/ --verbose` and got this output: https://gist.github.com/fenrhil/30a1115f012388fd10ee886dc8d75aa0

A possible reason why the bug was not fixed might be that git's error message is in my native language (French). Since [the fix](https://github.com/nirvdrum/svn2git/commit/5842a2a4570fe111ca26e39b153a651be3358480) seems to rely on git's error messages, maybe svn2git should take care of setting git's language before executing commands?
From [this google search result](http://askubuntu.com/questions/320661/change-gits-language-to-english-without-changing-the-locale), I gather that one can change git's language by setting the `LANG`, `LC_ALL` or `LANGUAGE` environment variables to something like "C".
I just tested running `LANG=C svn2git svn://svn.code.sf.net/p/jade-plugins/code/ --verbose` and it does seem to work just fine: https://gist.github.com/fenrhil/896e2e73bfcf764826070f97fc0dace5

**Comment by DanielRuf** [comment]
> With git 1.9.1

Isn't there Git 2.8 already? Git 1.9.1 is from 18-Mar-2014.

See here for a working workaround (should fix this): https://github.com/nirvdrum/svn2git/issues/132#issuecomment-31453436

References, that also Git 1.9.x is affected:

https://github.com/nirvdrum/svn2git/issues/132#issuecomment-41787321
https://github.com/nirvdrum/svn2git/issues/132#issuecomment-43014391
https://github.com/nirvdrum/svn2git/issues/132#issuecomment-43014391

This was changed in Git 1.8.x
https://github.com/nirvdrum/svn2git/issues/132#issuecomment-43163527

This is still a bug in svn2git which has to use the new syntax for tracking branches and which was changed in Git 1.8.x

So instead of using an outdated and insecure version of Git, just fix the mentioned file in svn2git and it should be fine.

I hope this fixes your issue.

**Comment by fenrhil** [comment]
@DanielRuf Thanks for you answer.

Git 1.9.1 is the current version of Git on Ubuntu LTS (trusty). But the new LTS will be released in the following days, as every 18 months, so yes, it's really old. See http://packages.ubuntu.com/fr/trusty/git

As for the issue:

 * your solution should indeed fix the problem
 * it seems a fix (that closely resemble your solution) has actually been committed here: https://github.com/nirvdrum/svn2git/commit/5842a2a4570fe111ca26e39b153a651be3358480 and released in 2.3.1, as mentioned in message https://github.com/nirvdrum/svn2git/issues/132#issuecomment-43168470
 * *but* this fix relies on the user's system language: it parses git's error messages so as to switch to an alternative behavior when a specific error is encountered (as far as I can tell)
 * *so* I tried changing environment variables (in this case, `LANG=C`) before running svn2git, and it "fixes the fix" for me.

So, to sum up, everything fine with both the fix released in 2.3.1 and the workaround of changing the environment variable. I was just warning the developer (@nirvdrum?) that his fix does not seem to work in non-english environments.

**Comment by DanielRuf** [comment]
Possibly makes sense to open a new issue for this problem related to the system language. The proposed fix using `LANG=C` on the cli could be helpful for other users facing the same problem.

But even if my system language is in German I use Git in English and so I never faced or noticed this problem, I'm fine with English ;-)

The possible usage of `LANG=C` could be mentioned in the readme.

It seems this issue is still in the master branch, at least I got no result when I tried grepping for `lang` and so on.

I think @nirvdrum is quite busy with JRuby and development for svn2git stalled. There are still many open PRs and issues.

Documenting this as a separate issue with the workaround using `LANG=C` could be quite helpful for other users who stumple upon this here.

**Comment by mazong1123** [comment]
I've resolved this issue within a new release in [svn2git.net](https://github.com/mazong1123/svn2gitnet/). In case anyone wants it, just download the latest release from https://github.com/mazong1123/svn2gitnet/releases
